### PR TITLE
Add tables, new record and queries to fetch data from all the tables.

### PR DESCRIPTION
**Pull request by jchernandez87**
### Changes: 😃 

- Create vets, specializations, and visits tables.
- Insert records into the new tables.
- Add queries to fetch data using references and join from the tables.

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -68,3 +68,30 @@ SELECT animals.name FROM animals JOIN owners ON animals.owner_id = owners.id WHE
...
+
+-- Get the latest animals visited by William Tatcher
+SELECT animals.name FROM animals JOIN visits ON animals.id = animal_id JOIN vets ON visits.vet_id = vets.id WHERE vets.name = 'William Tatcher' ORDER BY visits.date_of_visit DESC LIMIT 1;
+
+-- Stephanie mendez num of visited animals
+SELECT COUNT(animals.name) FROM animals JOIN visits ON animals.id = animal_id JOIN vets ON visits.vet_id = vets.id WHERE vets.name = 'Stephanie Mendez';
+
+-- List vets and their specializations
+SELECT vets.name, species.name FROM vets LEFT JOIN specializations ON vets.id = vet_id LEFT JOIN species ON species.id = species_id;
+
```
Comment (Emmyn5600) [inline queries.sql:80]: - [ ] Here we need to List vets and their specializations, so vets.id should be equal specializations.vet_id and specializations.species_id = species.id; in order to list both vets and their specializations.

File: schema.sql [hunk 2]
```diff
@@ -39,3 +39,29 @@ ALTER TABLE animals ADD CONSTRAINT fk_species_id FOREIGN KEY (species_id) REFERE
...
+);
+
+-- Create specializations join table referencing vets and species
+CREATE TABLE specializations (
+  species_id INT, 
+  vet_id INT, 
+  FOREIGN KEY (species_id) REFERENCES species (id) ON DELETE RESTRICT ON UPDATE CASCADE, 
+  FOREIGN KEY (vet_id) REFERENCES vets (id) ON DELETE RESTRICT ON UPDATE CASCADE, 
+  PRIMARY KEY (species_id, vet_id)
+);
```
Comment (Emmyn5600) [inline schema.sql:58]: - [ ] You should add a new join table with the name specializations using columns `species_id and vet_id`

File: schema.sql [hunk 3]
```diff
@@ -39,3 +39,29 @@ ALTER TABLE animals ADD CONSTRAINT fk_species_id FOREIGN KEY (species_id) REFERE
...
+
+-- Create visits join table referencing vets and animals
+CREATE TABLE visits (
+  date DATE,
+  animal_id INT, 
+  vet_id INT, 
+  FOREIGN KEY (animal_id) REFERENCES animals (id) ON DELETE RESTRICT ON UPDATE CASCADE, 
+  FOREIGN KEY (vet_id) REFERENCES vets (id) ON DELETE RESTRICT ON UPDATE CASCADE, 
+);
```
Comment (Emmyn5600) [inline schema.sql:67]: - [ ] You should add a new join table with the name visits with the following columns.
   animal_id int,
   vet_id int,
   visit_date date -

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUESTED :recycle:

Hi, @jchernandez87 

Great work on implementing the project requirements, however there are some issues to be solved before you get approved but you are almost there:muscle:
   
 - The repo looks professional.:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Changes required :writing_hand: 

         - Please work on the review too

    Cheers and Happy coding!:clap::clap::clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: data.sql [hunk 4]
```diff
@@ -41,3 +41,39 @@ UPDATE animals SET owner_id = 4 WHERE name = 'Charmander' OR name = 'Squirtle' O
...
+INSERT INTO vets (name, age, date_of_graduation) VALUES ('William Tatcher', 45, 'Apr 3,2000');
+INSERT INTO vets (name, age, date_of_graduation) VALUES ('Maisy Smith', 26, 'Jan 17,2019');
+INSERT INTO vets (name, age, date_of_graduation) VALUES ('Stephanie Mendez', 64, 'May 4,1981');
+INSERT INTO vets (name, age, date_of_graduation) VALUES ('Jack Harkness', 38, 'Jun 8,2008');
+
+-- Speciolizations records insert
+INSERT INTO specializations (species_id, vet_id) SELECT species.id, vets.id FROM species JOIN vets ON species.name = 'Pokemon' AND vets.name = 'William Tatcher';
+INSERT INTO specializations (species_id, vet_id) SELECT species.id, vets.id FROM species JOIN vets ON species.name = 'Pokemon' AND vets.name = 'Stephanie Mendez';
+INSERT INTO specializations (species_id, vet_id) SELECT species.id, vets.id FROM species JOIN vets ON species.name = 'Digimon' AND vets.name = 'Stephanie Mendez';
+INSERT INTO specializations (species_id, vet_id) SELECT species.id, vets.id FROM species JOIN vets ON species.name = 'Digimon' AND vets.name = 'Jack Harkness';
```
[the remark this answers is not in the document]
Comment (jchernandez87) [reply data.sql:56]: Hi!! Emmyn. I choose to use the INSERT INTO SELECT STATMENT to assign the values.

insertInto

And here is how my table looks:

Specializations

File: data.sql [hunk 5]
```diff
@@ -41,3 +41,39 @@ UPDATE animals SET owner_id = 4 WHERE name = 'Charmander' OR name = 'Squirtle' O
...
+INSERT INTO visits (animal_id, vet_id, date_of_visit) SELECT animals.id, vets.id, 'Jan 24, 2019' FROM animals JOIN vets ON animals.name = 'Boarmon' AND vets.name = 'Maisy Smith';
+INSERT INTO visits (animal_id, vet_id, date_of_visit) SELECT animals.id, vets.id, 'May 25, 2019' FROM animals JOIN vets ON animals.name = 'Boarmon' AND vets.name = 'Maisy Smith';
+INSERT INTO visits (animal_id, vet_id, date_of_visit) SELECT animals.id, vets.id, 'Feb 27, 2020' FROM animals JOIN vets ON animals.name = 'Boarmon' AND vets.name = 'Maisy Smith';
+INSERT INTO visits (animal_id, vet_id, date_of_visit) SELECT animals.id, vets.id, 'Aug 3, 2020' FROM animals JOIN vets ON animals.name = 'Boarmon' AND vets.name = 'Maisy Smith';
+INSERT INTO visits (animal_id, vet_id, date_of_visit) SELECT animals.id, vets.id, 'May 24, 2020' FROM animals JOIN vets ON animals.name = 'Blossom' AND vets.name = 'Stephanie Mendez';
+INSERT INTO visits (animal_id, vet_id, date_of_visit) SELECT animals.id, vets.id, 'Jan 11, 2021' FROM animals JOIN vets ON animals.name = 'Blossom' AND vets.name = 'William Tatcher';
```
[the remark this answers is not in the document]
Comment (jchernandez87) [reply data.sql:78]: For this one, I did the same.
And here is how my table looks:

Visits

File: queries.sql (hunk 1 above)
Comment (jchernandez87) [reply queries.sql:80]: Thank !!,  I miss this one.

File: schema.sql (hunk 2 above)
Comment (jchernandez87) [reply schema.sql:58]: I'm confused about this one. As I read in the documentation, the way you create a join table is by referencing the other tables.
I create the table following the documentation.

many

Join

Comment (Emmyn5600) [reply schema.sql:58]: I found that, this shouldn't be an issue, kindly work on the other changes requested.

File: schema.sql (hunk 3 above)
Comment (jchernandez87) [reply schema.sql:67]: I have the same doubt about this one.

Here is how my table looks:

Visits

File: data.sql (hunk 4 above)
[the remark this answers is not in the document]
Comment (Emmyn5600) [reply data.sql:56]: nice

File: data.sql (hunk 5 above)
[the remark this answers is not in the document]
Comment (Emmyn5600) [reply data.sql:78]: nice

File: queries.sql (hunk 1 above)
Comment (Emmyn5600) [reply queries.sql]: Good job

**Review by Emmyn5600 — APPROVED** [review]
Hi,

Good job so far! 👏🏼 @jchernandez87 

May I highlight the following:

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:
Your project is complete!  It's time to merge it :shipit:
Great work on implementing the changes requested by the previous reviewer.

 #### Optional suggestions
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Pull request merged**